Liferea/Liferea

```diff
@@ -24,137 +24,164 @@
-
-    def get_number_of_new_msgs(self):
-        self.number_of_new_msgs = self.liferea.GetNewItems()
-
-    def set_offline(self):
-        self.liferea.SetOnline(False)
-        self.set_offline_icon()
-
-    def set_offline_icon(self):
-        self.icon.SetIcon(os.path.abspath("./data/offline.png"))
-
-    def set_online(self):
-        self.liferea.SetOnline(True)
-        self.set_online_icon()
-
-    def set_online_icon(self):
-        self.icon.SetIcon(os.path.abspath("./data/online.png"))
-
-    def subscribe(self, url):
-        self.liferea.Subscribe(url) 
-
-    def switch_window_focus(self):
-        if not self.icon.Get("has_focus"):                                          # liferea running, but its window has not the focus
-            try:                                                                    
-                self.icon.ActOnAppli("show")                                           # bring the window to the foreground
-            except:                                                                 # if liferea is in the notification area, its window is actually closed, so we must re-open it
-                subprocess.Popen("liferea")                                         # ... so it is necessary to recall the application
-        else:
-            self.icon.ActOnAppli("minimize")                                              # take the window to the background
-
-    def show_icon_information(self):
-        if self.config_info == self.show_number_of_unread_msgs:
-            self.icon.SetQuickInfo(str(self.number_of_unread_msgs))
-        elif self.config_info == self.show_number_of_new_msgs:
-            self.icon.SetQuickInfo(str(self.number_of_new_msgs))
+  def __init__(self):
+      
+    self.liferea = None
+    self.number_of_unread_msgs = 0
+    self.number_of_new_msgs = 0
+    self.show_number_of_unread_msgs = 0
+    self.show_number_of_new_msgs = 1
+
+    self.set_status_online_key = 1
+    self.set_status_offline_key = 2
+    self.refresh_key = 3
+
+    self.dialog_active_time = 30                                                    # time in seconds that the dialog window will be active
+    CDApplet.__init__(self)                                                         # call high-level init
+
+  def connect_to_liferea(self):
+    while not self.liferea:
+      try:
+        bus = dbus.SessionBus()
+        liferea_object = bus.get_object('org.gnome.feed.Reader', '/org/gnome/feed/Reader')
+        self.liferea = dbus.Interface(liferea_object, dbus_interface='org.gnome.feed.Reader')
+      except dbus.DBusException:
+        # Something more elaborated could be done here, e.g., use gobject.idle_add and using,
+        # as callback, a method that would keep checking if Lirefea is open. Opted for the simpler approach
+        self.icon.PopupDialog({'message':'Impossible connect to Liferea, check if it is running. Retrying ...', 'time-length':self.dialog_active_time},{})
+        sleep(1)
```

Comment:
But then, if it's not possible to get Liferea's DBus interface, we'll get a popup each second. It can be annoying :-)

Is it not possible to be notified when the bus is available?